ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingAroundAngleBracketsRule.kt

```diff
@@ -0,0 +1,54 @@
+package com.pinterest.ktlint.ruleset.experimental
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.GT
+import com.pinterest.ktlint.core.ast.ElementType.LT
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_ARGUMENT_LIST
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import com.pinterest.ktlint.core.ast.nextLeaf
+import com.pinterest.ktlint.core.ast.prevLeaf
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+class SpacingAroundAngleBracketsRule : Rule("angle-brackets-rule") {
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        if (node.elementType == TYPE_ARGUMENT_LIST) {
+            val openingBracketChild = node.nextLeaf { it.elementType == LT }
+            if (openingBracketChild != null) {
+                // Check for rogue spacing before an opening bracket:  Map <String, Int>
+                val beforeAngleBracket = openingBracketChild.prevLeaf()
+                if (beforeAngleBracket != null && beforeAngleBracket.elementType == WHITE_SPACE) {
```

Comment:
I think this can be simplified to `beforeAngleBracket?.elementType == WHITE_SPACE`, also in other such `ifs`. If you don't have special conventions for that, of course.